internal/handlers/handlers.go

```diff
@@ -1 +1,58 @@
 package handlers
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Yandex-Practicum/go1fl-sprint6-final/internal/service"
+)
+
+func UploadHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	file, header, err := r.FormFile("myFile")
+	if err != nil {
+		http.Error(w, "Failed to read file", http.StatusInternalServerError)
+		return
+	}
+	defer file.Close()
+
+	data, err := io.ReadAll(file)
+	if err != nil {
+		http.Error(w, "Failed to read file content", http.StatusInternalServerError)
+		return
+	}
+
+	input := string(data)
+	output, err := service.AutoDetectAndConvert(input)
+	if err != nil {
+		http.Error(w, "Conversion failed", http.StatusInternalServerError)
+		return
+	}
+
+	filename := fmt.Sprintf("result_%s%s", time.Now().Format("20060102150405"), filepath.Ext(header.Filename))
+	if err := os.WriteFile(filename, []byte(output), 0644); err != nil {
+		http.Error(w, "Failed to save result", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/plain")
```

Comment:
Чтобы не было кракозябр в браузере нужно так писать
```suggestion
	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
```